runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -781,6 +781,7 @@ J9::Z::TreeEvaluator::inlineVectorizedStringIndexOf(TR::Node* node, TR::CodeGene
       generateRRInstruction(cg, TR::InstOpCode::getAddRegOpCode(), node, matchIndexReg, stringIndexReg);
       cursor = generateRIEInstruction(cg, TR::InstOpCode::getCmpImmBranchRelOpCode(), node, patternLenReg, (int8_t)vectorSize, labelLoadResult, TR::InstOpCode::COND_BNH);
       iComment("if patternLen <= 16 then we are done, otherwise we continue to check the rest of pattern");
+      generateRIEInstruction(cg, TR::InstOpCode::getCmpRegAndBranchRelOpCode(), node, stringIndexReg, maxIndexReg, labelPatternNotFound, TR::InstOpCode::COND_BH);
```

Comment:
Similarly, an iComment maybe useful here. i.e. something like: "Updated stringIndex for start of matched section exceeds maxIndex. Full pattern cannot be matched."

Reply:
Fixed in [force-push](https://github.com/eclipse-openj9/openj9/compare/b2389b0722853a5cac394cbb8948c6e8ebdc9cdc..c20200d).